### PR TITLE
light: Switch to native logger

### PR DESCRIPTION
**Pull request by mitzkia**
For now internal classes: Logger and LoggerFactory are strong dependencies inside the lib. They are passed for those classes where we want to print some log messages, but they are not used everywhere.
Pytest also has a good support for using native logging mechanism. We can easily add support for this.
Until now there was one missing feature from pytest side namely: it could not create report file (log-file) dynamically (from hook). This has been changed, there is an upcoming PR: https://github.com/pytest-dev/pytest/pull/4752

Update (2019-02-22):
* New pytest releated PR: https://github.com/pytest-dev/pytest/pull/4812

Missing steps:
- [x] save AssertionErrors also into report files (as earlier)
- [x] check created logger objects in classes/modules do they really needed
- [x] remove --loglevel starter option

Decision needed:
- about logging format (example): `%(asctime)s:%(msecs)d - %(filename)s:%(lineno)d->%(funcName)s - %(levelname)s - %(message)s`

WIP state:
- This PR should be in WIP state until the pytest's PR is not merged and released
- TravisCI is failing because of this

How logging will work:
1, setup initial logging parameters via command line or pytest.ini
2, Add `import logging` into module where you want to print some log
3, create new logger for that class or module
4, use created logger with a proper debug level

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: tests/python_functional/conftest.py [hunk 1]
```diff
@@ -108,3 +108,19 @@ def pytest_runtest_logreport(report):
     logger = logging.getLogger("Conftest")
     if report.outcome == "failed":
         logger.error("\n{}".format(report.longrepr))
+
+
+def construct_report_file_path(item):
+    absolute_framework_dir = Path.cwd()
+    relative_report_dir = item._request.getfixturevalue("reports")
```
Comment (furiel) [inline tests/python_functional/conftest.py]: I feel this a little limiting. We could allow reports to be absolute. For example with
```
def construct_report_file_path(item):
    report_dir = Path(item._request.getfixturevalue("reports"))
    testcase_name = item._request.node.name
    file_name = "testcase_{}.log".format(testcase_name)
    return Path(report_dir, testcase_name, file_name).absolute()
```

Using `.absolute()` removes the manual work for calculating `absolute_framework_dir`, and should work both with absolute and relative paths.

Comment (mitzkia) [reply tests/python_functional/conftest.py]: +1, and fixed

**Comment by mitzkia** [comment]
Related to this PR:
* We are still waiting a new release of pytest with the following (merged) PR: https://github.com/pytest-dev/pytest/pull/4812
* Needed a decision about the default log-format (actual option available in a new pytest.ini file), my proposal is:
"22:23:22:747 - syslog_ng_cli.py:93->start - INFO - Beginning of syslog-ng start"
* Check the current internal log-messages, do they informative? are they on the proper log level? Is there any missing log message from the current lib?

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by mitzkia** [comment]
I have updated the PR:
* default log level is INFO
* actual log messages are rewritten/fixed

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: dbld/images/helpers/pip_packages.manifest [hunk 2]
```diff
@@ -6,7 +6,6 @@ ply                     [centos, debian, ubuntu]
 pylint                  [centos-7, debian, ubuntu]
 
 # pip packages for python functional tests
-colorlog                [centos, debian, ubuntu]
```
Comment (furiel) [inline dbld/images/helpers/pip_packages.manifest]: This first patch does not stand on its own. It breaks bisectibility. Can you move to the last commit?

File: tests/python_functional/src/executors/command_executor.py [hunk 3]
```diff
@@ -72,16 +67,11 @@ def run(self, command, stdout_path, stderr_path):
     def __process_std_outputs(self, stdout, stderr):
         stdout_content = stdout.open_file("r").read()
         stderr_content = stderr.open_file("r").read()
-        self.__logger.debug("Stdout:[{}]".format(stdout_content))
```
Comment (furiel) [inline tests/python_functional/src/executors/command_executor.py]: These logs seem useful. Do we need to delete these? Maybe putting a lower loglevel?

File: dbld/images/helpers/pip_packages.manifest (hunk 2 above)
Comment (mitzkia) [reply dbld/images/helpers/pip_packages.manifest]: +1

File: tests/python_functional/src/executors/command_executor.py (hunk 3 above)
Comment (mitzkia) [reply tests/python_functional/src/executors/command_executor.py]: +1

**Comment by mitzkia** [comment]
I have fixed both comments, and pushed.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**